astropy/cosmology/tests/test_connect.py

```diff
@@ -32,6 +33,14 @@
                   ("astropy.cosmology", Cosmology),
                   ("astropy.row", Row), ("astropy.table", QTable)]
 
+# Checks if bs4 can be accessed
+bs4Imported = True
+try:
+    from bs4 import BeautifulSoup
+except ImportError:
+    bs4Imported = False
```

Comment:
Hm. are you sure ``HAS_BS4`` isn't enough?

Reply:
If it isn't, this should be checked in the imports section and incorporated into  ``readwrite_formats``, not added as a separate flag in the tests.

```python
readwrite_formats = ... ("ascii.html", False, HAS_BS4 & bs4Imported), ...
```

Reply:
But first try just ``HAS_BS4``.

Reply:
I was trying having only HAS_BS4 and the combination of HAS_BS4 and bs4Imported, but both seem to fail. I get a feeling that I am missing a `pytest.skip` somewhere, but I am not able to locate where the issue still occurs. I thought that HAS_BS4 might not be working, so that was why I added the bs4Imported later, but there still seems to be issues with the bs4 library not being included. 

Although, after I added HAS_BS4, there were less errors reported in the CI.

Reply:
I looked in the CI traceback for the source of the error. The traceback shows all the function calls that leads to an error, so it's very useful for locating the source. The errors are now coming from ``test_html.py``, which doesn't have any skips if ``bs4`` is missing.

I think all the methods of ``ReadWriteHTMLTestMixin`` will fail if ``bs4`` is missing, so each can be decorated with the decorator ``@pytest.mark.skipif(not HAS_BS4, reason="requires bs4")``

Reply:
Oh, I understand now. I will clean up the code after modifying per your suggestions. Thanks!